src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs

```diff
@@ -104,6 +104,38 @@ private static ImmutableHashSet<string> GetSuppressedTypeNames(Compilation compi
                 .ToImmutableHashSet();
         }
 
+        /// <summary>
+        /// Add some additional files into this project
+        /// </summary>
+        /// <param name="directory"></param>
+        /// <param name="skipPredicate"></param>
+        /// <param name="folders"></param>
+        public void AddDirectory(string directory, Predicate<string>? skipPredicate = null, IEnumerable<string>? folders = null)
+        {
+            _project = AddDirectory(_project, directory, skipPredicate, folders);
+        }
+
+        /// <summary>
+        /// Add the files in the directory to a project per a given predicate with the folders specified
+        /// </summary>
+        /// <param name="project"></param>
+        /// <param name="directory"></param>
+        /// <param name="skipPredicate"></param>
+        /// <param name="folders"></param>
+        /// <returns></returns>
+        internal static Project AddDirectory(Project project, string directory, Predicate<string>? skipPredicate = null, IEnumerable<string>? folders = null)
```

Comment:
Let's use `Func<string, bool>` instead of `Predicate<string>`

Reply:
Fixed